src/testcases/org/apache/poi/ss/formula/TestWorkbookEvaluator.java

```diff
@@ -606,4 +608,38 @@ private void testIFEqualsFormulaEvaluation_evaluateAllFormulaCells(
         
         testIFEqualsFormulaEvaluation_teardown(wb);
     }
+
+    @Test
+    public void testSheetNameReference() throws IOException {
+        Workbook wb = new HSSFWorkbook();
+
+        Sheet s1 = wb.createSheet("Sheet1");
+        Cell cell1 = CellUtil.getCell(CellUtil.getRow(0, s1), 0);
+        cell1.setCellValue(123);
+        Sheet s2 = wb.createSheet();
+        Cell cell2 = CellUtil.getCell(CellUtil.getRow(0, s2), 0);
+
+        FormulaEvaluator eval = wb.getCreationHelper().createFormulaEvaluator();
+        {
+            cell2.setCellFormula("Sheet1!A1");
+            CellValue value = eval.evaluate(cell2);
+            assertEquals(123d, value.getNumberValue(), 0);
+        }
+
+        for (int i = 256; i <= 65535; i++) {
+            char c = (char) i;
+            String name = String.valueOf(c);
+            try {
+                WorkbookUtil.validateSheetName(name);
+            } catch (IllegalArgumentException e) {
+                continue;
```

Comment:
Is this test considered failing if an IllegalArgumentException is ever raised by validateSheetName here?
If so:

    } catch (IllegalArgumentException e) {
        fail("Expected sheet validation to pass, but it failed with a sheet name of " + name);
    }

Or is this more the case of "assuming that name is a valid sheet name, verify that Sheet 0's name can be set to name, and a cell formula can be set and evaluated"?

A unit test should clearly define what should pass and what should fail. If setSheetName, setCellFormula, and evaluate call WorkbookUtil.validateSheetName, then this unit test is a tautology. It will always pass and therefore should be pruned from the unit test suite to eliminated wasted test time.